Signal
---------------------------

В сигнале 

    from django.core.cache import cache
    
    @receiver(m2m_changed, sender=Operation.over_printsheet.through)
    def after_operation_save_sheet(sender, instance: Operation, action, **kwargs):
        if action == 'post_add':

            try:
                instance.some_method()
            except Exception as e:
                cache.set('signal_message', str(e), timeout=60)


  Если у нас неудачно завершилось какое-то действие, добавляем в кеш запись `cache.set('name_of_action', error_string, timeout=60)`

  Во view, которая потенциально вызывает этот сигнал (при каком-то действии пользователя), обрабатываем эту запись из кеша:
    
    if signal_error_message:
        messages.error(request, signal_error_message)
        response = HttpResponse(status=406)  # or other error code
    else:
        messages.success(request, 'Update success!')
        response = HttpResponse(status=200)    

    return response
